web/src/message_list.ts

```diff
@@ -125,19 +138,26 @@ export class MessageList {
         return true;
     }
 
-    is_current_message_list() {
+    is_current_message_list(): boolean {
         return this.view.is_current_message_list();
     }
 
-    prevent_reading() {
+    prevent_reading(): void {
         this.reading_prevented = true;
     }
 
-    resume_reading() {
+    resume_reading(): void {
         this.reading_prevented = false;
     }
 
-    add_messages(messages, opts) {
+    add_messages(
+        messages: Message[],
+        append_to_view_opts: {messages_are_new?: boolean},
+    ):
+        | {
+              need_user_to_scroll: boolean;
+          }
```

Comment:
This is the `RenderInfo` type (which should be moved from `message_lists.ts` to this file as well).

Reply:
Updated and moved